src/NuGet.Core/NuGet.PackageManagement/Utility/ExceptionUtility.cs

```diff
@@ -12,20 +12,12 @@ public static Exception Unwrap(Exception exception)
         {
             if (exception == null)
             {
-                throw new ArgumentNullException("exception");
+                throw new ArgumentNullException(nameof(exception));
             }
 
-            if (exception.InnerException == null)
+            while (exception.InnerException != null)
             {
-                return exception;
-            }
-
-            // Always return the inner exception from a target invocation exception
```

Comment:
Why did you remove this?

Reply:
this is incorrect for my change, It always return the root cause exception from any exception, not just invocation exception, also not the inner exception